tests/applications/test_turn_channels_api.py

```diff
@@ -0,0 +1,520 @@
+                async with handle_inbound(tca_worker, msg):
+                    await http_server.receive_req()
+                    await http_server.send_rsp(RspInfo(code=502, wait=0.3))
+
+    [err] = [log for log in caplog.records if log.levelno >= logging.ERROR]
+    assert "Timed out sending message after 0.2 seconds." in err.getMessage()
+    assert e.value.name == "TimeoutError"
+    assert e.value.description == "timeout"
+    assert e.value.status == HTTPStatus.BAD_REQUEST
+
+
+async def test_inbound_auth_token(worker_factory, http_server):
+    """
+    If mo_message_url_auth_token is set, we use its token in the
+    Authorization header.
+    """
+    token = "my-token"  # noqa: S105 (This is a fake token.)
+    config = mk_config(http_server, auth_token=token)
+
+    msg = mkmsg("hello")
+
+    async with worker_factory.with_cleanup(TurnChannelsApi, config) as tca_worker:
+        await tca_worker.setup()
+        with fail_after(2):
+            async with handle_inbound(tca_worker, msg):
+                req = await http_server.receive_req()
+                await http_server.send_rsp(RspInfo())
+
+    assert req.headers["Authorization"] == "Bearer my-token"
+
+
+async def test_forward_ack_bad_response(tca_worker, http_server, caplog):
+    """
+    If forwarding an ack results in an HTTP error, the error and event
+    are logged.
+    """
+    ev = mkev("msg-21", EventType.ACK)
+
+    with fail_after(2):
+        async with handle_event(tca_worker, ev):
+            await http_server.receive_req()
+            await http_server.send_rsp(RspInfo(code=500))
+
+    [err] = [log for log in caplog.records if log.levelno >= logging.ERROR]
+    assert "Error sending event, received HTTP code 500" in err.getMessage()
+
+
+async def test_forward_ack_too_slow(worker_factory, http_server, caplog):
+    """
+    If forwarding an ack times out, the error and message are logged.
+
+    Using an autojump clock here seems to break the AMQP client, so we
+    have to use wall-clock time instead.
+    """
+    config = mk_config(http_server, event_url_timeout=0.2)
+
+    async with worker_factory.with_cleanup(TurnChannelsApi, config) as tca_worker:
+        await tca_worker.setup()
+        ev = mkev("msg-21", EventType.ACK)
+        with fail_after(5):
+            async with handle_event(tca_worker, ev):
+                await http_server.receive_req()
+                await http_server.send_rsp(RspInfo(code=502, wait=0.4))
+
+    [err] = [log for log in caplog.records if log.levelno >= logging.ERROR]
+    assert "Timed out sending event after 0.2 seconds." in err.getMessage()
+
+
+async def test_forward_nack(worker_factory, http_server):
+    """
+    A nack event referencing an outbound message we know about is
+    forwarded over HTTP.
+    """
+    ev = mkev("msg-21", EventType.NACK, nack_reason="KaBooM!")
+
+    config = mk_config(http_server, turn_base_url_path=f"{http_server.bind}/event")
```

Comment:
So here the base url should just be `http_server.bind`, and the path should be added by our application, depending on the type of thing we're sending. If it's a message, it should be `/messages`, if it's an event (ack/nack/dr), it should be `/statuses`